slick/src/main/scala/slick/jdbc/MacroTreeBuilder.scala

```diff
@@ -74,13 +73,9 @@ private[jdbc] class MacroTreeBuilder[C <: Context](val c: C)(paramsList: List[C#
     })
 
     resultTypes.size match {
-      case 0 => implicitTree(TypeTree(typeOf[Int]) , GetResultTypeTree)
-      case 1 => implicitTree(resultTypeTrees(0), GetResultTypeTree)
-      case n if (n <= 22) =>
-        implicitTree(AppliedTypeTree(
-          Select(Select(Ident(termNames.ROOTPKG), TermName("scala")), TypeName("Tuple" + resultTypes.size)),
-          resultTypeTrees.toList
-        ), GetResultTypeTree)
+      case 0 => q"_root_.scala.Predef.implicitly[_root_.slick.jdbc.GetResult[Int]]"
+      case 1 => q"_root_.scala.Predef.implicitly[_root_.slick.jdbc.GetResult[${resultTypeTrees(0)}]]"
+      case n if (n <= 22) => q"_root_.scala.Predef.implicitly[_root_.slick.jdbc.GetResult[(..$resultTypeTrees)]]"
```

Comment:
aren't we cross-compiling slick to 2.10? Quasi-quotes would require macro-paradise then. @szeiger any opinion on that?